opentreemap/treemap/js/src/lib/mapPage.js

```diff
@@ -44,6 +52,15 @@ module.exports.init = function (options) {
             searchBar.filterNonGeocodeObjectStream,
             geocodeEvents);
 
+    if (options.shouldUseLocationSearchUI) {
```

Comment:
Understanding what's happening with `programmaticallyUpdatedStream` and why is tricky. How about a comment here like:

When loading the page from a URL with a `BOUNDARY_ID` search we must wait for the location typeahead to populate the "Search by Location" input box (by fetching the boundary name from the server). That triggers an event on `searchBar.programaticallyUpdatedStream` and we call `locationSearchUI.onSearchChanged`, which uses the presence of text in the "Search by Location" input box to distinguish between named and anonymous boundaries.